### PR TITLE
Enable macOS build

### PR DESCRIPTION
**Pull request by hoechenberger**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by hoechenberger** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -35,8 +32,9 @@ requirements:
 
 test:
   commands:
-    - test -f $PREFIX/lib/libglfw.so               # [linux]
-    - conda inspect linkages -p $PREFIX $PKG_NAME  # [linux]
+    - test -f $PREFIX/lib/libglfw.so                         # [linux]
+    - test -f $PREFIX/lib/libglfw.dylib                      # [osx]
```
Comment (hmaarrfk) [inline recipe/meta.yaml]: there was some kind of macro, like {{ SHEXT }} for both unixes but meh.

**Comment by hmaarrfk** [comment]
lol, why did windows fail?

File: recipe/meta.yaml (hunk 1 above)
Comment (hoechenberger) [reply recipe/meta.yaml]: So … is it okay to keep this as it is?

**Comment by hoechenberger** [comment]
> lol, why did windows fail?

Seems to be an issue with the build system:
```
%SRC_DIR%>CALL "C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\..\..\VC\vcvarsall.bat" amd64  
CMake Error at CMakeLists.txt:3 (project):
  Failed to run MSBuild command:

    MSBuild.exe

  to get the value of VCTargetsPath:

    Microsoft (R) Build Engine version 4.8.3761.0
    [Microsoft .NET Framework, version 4.0.30319.42000]
    Copyright (C) Microsoft Corporation. All rights reserved.
    
    Build started 7/5/2019 2:15:22 PM.
    Project "D:\bld\glfw_1562336041991\work\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj" on node 1 (default targets).
    D:\bld\glfw_1562336041991\work\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "D:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    Done Building Project "D:\bld\glfw_1562336041991\work\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj" (default targets) -- FAILED.
    
    Build FAILED.
    
    "D:\bld\glfw_1562336041991\work\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj" (default target) (1) ->
      D:\bld\glfw_1562336041991\work\build\CMakeFiles\3.14.5\VCTargetsPath.vcxproj(15,2): error MSB4019: The imported project "D:\Microsoft.Cpp.Default.props" was not found. Confirm that the path in the <Import> declaration is correct, and that the file exists on disk.
    
        0 Warning(s)
        1 Error(s)
    
    Time Elapsed 00:00:00.31
    

  Exit code: 1
```

**Comment by hoechenberger** [comment]
There is also something odd about the macOS build, which I reported here: https://github.com/conda/conda-build/issues/3593

**Comment by hmaarrfk** [comment]
oh right it failed the last time too. But appveyor was good, so i just went with it. Sometimes Azure bugs out. Not too sure.

**Comment by hmaarrfk** [comment]
looks good. do you want to maintain this too? Add yourself to the bottom. I'll merge this later today when it passes.

**Comment by hoechenberger** [comment]
> oh right it failed the last time too. But appveyor was good, so i just went with it. Sometimes Azure bugs out. Not too sure.

So… shall we go ahead and get this merged? And I can then check whether the macOS build is working properly on an "actual" machine. Not sure about the potential linking problem!

**Comment by hoechenberger** [comment]
> looks good. do you want to maintain this too? Add yourself to the bottom. I'll merge this later today when it passes.

Yeah, I can do that :) Thanks!

File: recipe/meta.yaml [hunk 2]
```diff
@@ -35,8 +32,7 @@ requirements:
 
 test:
   commands:
-    - test -f $PREFIX/lib/libglfw.so               # [linux]
-    - conda inspect linkages -p $PREFIX $PKG_NAME  # [linux]
```
Comment (hoechenberger) [inline recipe/meta.yaml]: Sure you _wanted_ to remove this line too?

Comment (hmaarrfk) [reply recipe/meta.yaml]: Yeah I think cons build does this automatically now

**Comment by hmaarrfk** [comment]
Are you sure osx doesn't have pkg config

**Comment by hmaarrfk** [comment]
Thanks!

**Pull request merged**